### ISSUE TITLE
Unexpected Exception: 'results' (ansible/plugins/strategy/linear.py)

### ISSUE THREAD

**Issue by rrey** [ansible/ansible#21590]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
plugin

##### ANSIBLE VERSION
ansible 2.2.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides

##### CONFIGURATION
roles_path    = /etc/ansible/roles:/home/remi/dev/project/common/ansible/roles

##### OS / ENVIRONMENT
debian 7.9

##### SUMMARY
When running the task below, I obtain the following error:
```
ERROR! Unexpected Exception: 'results'
the full traceback was:
Traceback (most recent call last):
  File "/usr/local/bin/ansible-playbook", line 103, in <module>
    exit_code = cli.run()
  File "/usr/local/lib/python2.7/dist-packages/ansible/cli/playbook.py", line 159, in run
    results = pbex.run()
  File "/usr/local/lib/python2.7/dist-packages/ansible/executor/playbook_executor.py", line 154, in run
    result = self._tqm.run(play=play)
  File "/usr/local/lib/python2.7/dist-packages/ansible/executor/task_queue_manager.py", line 282, in run
    play_return = strategy.run(iterator, play_context)
  File "/usr/local/lib/python2.7/dist-packages/ansible/plugins/strategy/linear.py", line 290, in run
    include_results = hr._result['results']
KeyError: 'results'
```

##### STEPS TO REPRODUCE

The task rasing the error is the following:

```
    - name: Install product packages 
      include_role: 
          name: push_and_install_rpm 
      vars: 
          package: "{{ item.name }}-{{ version }}*" 
          package_path: "{{ product_rpm_path }}/{{ item.arch }}" 
      with_items: "{{ packages }}" 
      when: 
          - include_product|bool 

```

To reproduce the issue, use the following variables:

```
include_product: no
packages:
    - { name: "whatever", arch: "x86_64" }
    - { name: "whatever", arch: "x86_64" }

```

##### EXPECTED RESULTS
With the provided reproduction step, the the "when" condition is false, so everything is skipped.
But once all items in packages are processed the exception is is raised.

I managed to reproduce the error with a simple debug message in the push_and_install_rpm role. that exclude the implication of the role in the failure.

##### ACTUAL RESULTS

```
TASK [Install product packages] ************************************************
task path: /home/remi/dev/project-master/utils/ansible/plays/deliver_project.yml:39
skipping: [testiris01] => (item={u'arch': u'x86_64', u'name': u'one-libs'})  => {
    "changed": false, 
    "item": {
        "arch": "x86_64", 
        "name": "one-libs"
    }, 
    "skip_reason": "Conditional check failed", 
    "skipped": true
}
skipping: [testiris01] => (item={u'arch': u'x86_64', u'name': u'one-core'})  => {
    "changed": false, 
    "item": {
        "arch": "x86_64", 
        "name": "one-core"
    }, 
    "skip_reason": "Conditional check failed", 
    "skipped": true
}
skipping: [testiris01] => (item={u'arch': u'x86_64', u'name': u'one-web'})  => {
    "changed": false, 
    "item": {
        "arch": "x86_64", 
        "name": "one-web"
    }, 
    "skip_reason": "Conditional check failed", 
    "skipped": true
}
skipping: [testiris01] => (item={u'arch': u'x86_64', u'name': u'one-integration'})  => {
    "changed": false, 
    "item": {
        "arch": "x86_64", 
        "name": "one-integration"
    }, 
    "skip_reason": "Conditional check failed", 
    "skipped": true
}
ERROR! Unexpected Exception: 'results'
the full traceback was:

Traceback (most recent call last):
  File "/usr/local/bin/ansible-playbook", line 103, in <module>
    exit_code = cli.run()
  File "/usr/local/lib/python2.7/dist-packages/ansible/cli/playbook.py", line 159, in run
    results = pbex.run()
  File "/usr/local/lib/python2.7/dist-packages/ansible/executor/playbook_executor.py", line 154, in run
    result = self._tqm.run(play=play)
  File "/usr/local/lib/python2.7/dist-packages/ansible/executor/task_queue_manager.py", line 282, in run
    play_return = strategy.run(iterator, play_context)
  File "/usr/local/lib/python2.7/dist-packages/ansible/plugins/strategy/linear.py", line 290, in run
    include_results = hr._result['results']
KeyError: 'results'
```

I stopped with pdb on the failing line : 
```
ipdb> hr._result
{'msg': u'All items completed', 'changed': False}
```

**Comment by bcoca** [comment]
that should have been fixed by #18545

**Comment by rrey** [comment]
That's indeed fixed by #18545

Thanks @bcoca
